File: golang/src/SharedKernel/Infrastructure/CommandBus/SimpleCommandBus.go
package CommandBus

import (
	. "../../Application/Command"
	. "../../Application/Command/Handler"
	"reflect"
)

type SimpleCommandBus struct {
	handlers map[string]Handler
}

func (commandBus *SimpleCommandBus) RegisterHandler(command Command, handler Handler) {
	commandBus.handlers = make(map[string]Handler)
	commandBus.handlers[reflect.TypeOf(command).String()] = handler
}

func (commandBus *SimpleCommandBus) Handle(command Command) error {
	return commandBus.handlers[reflect.TypeOf(command).String()].Handle(command)
}

File: python/src/sharedkernel/infrastructure/commandbus/SimpleCommandBus.py
from python.src.sharedkernel.application.command.Command import Command
from python.src.sharedkernel.application.command.handler.Handler import Handler
from python.src.sharedkernel.application.commandbus.CommandBus import CommandBus


class SimpleCommandBus(CommandBus):

    def __init__(self):
        self.__handlers = {}

    def register_handler(self, command: Command, handler: Handler):
        self.__handlers[command.name()] = handler

    def handle(self, command: Command):
        self.__handlers[command.name()].handle(command)
